c++/src/ColumnReader.cc

```diff
@@ -336,8 +336,7 @@ namespace orc {
           }
         }
         int64_t writerTime = secsBuffer[i] + epochOffset;
-        secsBuffer[i] = writerTime +
-          writerTimezone.getVariant(writerTime).gmtOffset;
+        secsBuffer[i] = writerTimezone.convertToUTC(writerTime);
```

Comment:
If it reads a file created by Java writer with writer timezone other than GMT, you are not going to get the same clock reading from ColumnPrinter, are you?